modules/iam-role-for-serviceaccount/main.tf

```diff
@@ -16,7 +16,7 @@ resource "aws_iam_role" "irsa" {
         Federated = var.oidc_arn
       }
       Condition = {
-        "${var.iam_conditional_operator}" = {
+        var.iam_conditional_operator = {
```

Comment:
Should use (var.iam_conditional_operator) is better syntax, without specific interpolation things get strange.